wallet/src/main/scala/org/bitcoins/wallet/internal/TransactionProcessing.scala

```diff
@@ -334,17 +325,30 @@ private[wallet] trait TransactionProcessing extends WalletLogger {
           logger.debug(
             s"Marked utxo=${updated.toHumanReadableString} as state=${updated.state}"))
         updatedF.map(Some(_))
-      case TxoState.Reserved | TxoState.PendingConfirmationsSpent |
-          BroadcastSpent =>
+      case TxoState.Reserved =>
         val updated =
-          out.copyWithSpendingTxId(spendingTxId)
-        val updatedF =
-          spendingInfoDAO.update(updated)
+          out
+            .copyWithSpendingTxId(spendingTxId)
+            .copyWithState(state = BroadcastSpent)
+        val updatedF = spendingInfoDAO.update(updated)
+        updatedF.map(Some(_))
+      case TxoState.BroadcastSpent =>
+        logger.warn(
+          s"Updating the spendingTxId of a transaction that is already spent, " +
+            s"old state=${TxoState.BroadcastSpent} old spendingTxId=${out.spendingTxIdOpt} new spendingTxId=${spendingTxId}")
+        val updated =
+          out
+            .copyWithSpendingTxId(spendingTxId)
+            //we need to go back to the BroadcastSpent state
+            //as we are overriding the previous spending tx
+            //therefore we can no longer use the old Txo state
+            .copyWithState(state = BroadcastSpent)
+        val updatedF = spendingInfoDAO.update(updated)
         updatedF.map(Some(_))
       case TxoState.ImmatureCoinbase =>
         Future.failed(new RuntimeException(
           s"Attempting to spend an ImmatureCoinbase ${out.outPoint.hex}, this should not be possible until it is confirmed."))
-      case TxoState.ConfirmedSpent =>
+      case TxoState.ConfirmedSpent | TxoState.PendingConfirmationsSpent =>
```

Comment:
I move this down here, i think we should treat these two as the same thing as it is overwriting a spendingTxId that has confirmations on it